terraform-provider-constellation/internal/provider/cluster_resource.go

```diff
@@ -904,6 +895,26 @@ func (r *ClusterResource) apply(ctx context.Context, data *ClusterResourceModel,
 	return diags
 }
 
+func (r *ClusterResource) getImageVersion(ctx context.Context, data *ClusterResourceModel) (imageAttribute, semver.Semver, diag.Diagnostics) {
+	var image imageAttribute
+	diags := data.Image.As(ctx, &image, basetypes.ObjectAsOptions{})
```

Comment:
We'd probably want to return early here, as when this fails (conversion error), the 2 subsequent calls are going to fail as well